src/app/vertextool/qgsvertextool.cpp

```diff
@@ -2602,7 +2602,9 @@ bool QgsVertexTool::matchEdgeCenterTest( const QgsPointLocator::Match &m, const
     return false;  // currently not supported for circular edges
 
   QgsRectangle visible_extent = canvas()->mapSettings().visibleExtent();
-  if ( !visible_extent.contains( p0 ) || !visible_extent.contains( p1 ) )
+  // Check if one point is inside and the other outside the visible extent in order to transpose the mid marker
+  // If both are inside or outside there is no such need
+  if ( !visible_extent.contains( p0 ) != !visible_extent.contains( p1 ) )
```

Comment:
```suggestion
  if ( visible_extent.contains( p0 ) != visible_extent.contains( p1 ) )
```

Same effect but easier to read